ibis/backends/tests/test_generic.py

```diff
@@ -2040,6 +2154,37 @@ def test_substitute(backend):
     assert expr["subs_count"].execute()[0] == t.count().execute() // 10
 
 
+@pytest.mark.broken("flink", reason="can't handle SELECT NULL AS `SearchedCase(None)`")
+def test_ibis_cases_empty(con):
+    assert pd.isnull(con.execute(ibis.cases()))
+    assert con.execute(ibis.cases(else_=2)) == 2
+    assert pd.isnull(con.execute(ibis.literal(1).cases()))
+    assert con.execute(ibis.literal(1).cases(else_=2)) == 2
+
+
+@pytest.mark.broken("clickhouse", reason="special case this and returns 'oops'")
+def test_value_cases_null(con):
+    """CASE x WHEN NULL never gets hit"""
+    e = ibis.literal(5).nullif(5).cases((None, "oops"), else_="expected")
+    assert con.execute(e) == "expected"
+
+
+@pytest.mark.broken("pyspark", reason="raises a ResourceWarning that we can't catch")
+def test_case(con):
+    # just to make sure that the deprecated .case() method still works
+    with pytest.warns(FutureWarning, match=".cases"):
```

Comment:
Each of these calls needs to be in a separate **single** `with pytest.warns` block.

Otherwise you're only asserting that one of the many lines in the block warns.